json-encode.go

```diff
@@ -16,7 +16,8 @@ import (
 
 // *** Encoding/MarshalJSON ***
 func (cdc *Codec) encodeReflectJSON(w io.Writer, info *TypeInfo, rv reflect.Value, opts FieldOptions) error {
-	disambiguate := rv.Kind() != reflect.Interface && (info.Registered || info.AlwaysDisambiguate)
```

Comment:
info.Registered implies rv.Kind() != reflect.Interface.
info.AlwaysDisambiguate is not necessary to check because our JSON spec says that all registered concrete types must include disambiguation...

Reply:
Gotcha, thank you for disambiguating that for me(pun unintended). Could you point me to our spec?

Reply:
It's not written yet :) I mean what we discussed on the phone a while back.

Reply:
Ack